common/attest_plugin.c

```diff
@@ -192,33 +193,67 @@ oe_result_t oe_verify_evidence(
     oe_result_t result = OE_UNEXPECTED;
     oe_plugin_list_node_t* plugin_node;
     oe_verifier_t* verifier;
-    oe_attestation_header_t* evidence =
-        (oe_attestation_header_t*)evidence_buffer;
-    oe_attestation_header_t* endorsements =
-        (oe_attestation_header_t*)endorsements_buffer;
-
-    if (!evidence_buffer || evidence_buffer_size < sizeof(*evidence) ||
-        (endorsements_buffer &&
-         endorsements_buffer_size < sizeof(*endorsements)))
+    const uint8_t* plugin_evidence = NULL;
+    size_t plugin_evidence_size = 0;
+    const uint8_t* plugin_endorsements = NULL;
+    size_t plugin_endorsements_size = 0;
+
+    if (!evidence_buffer || !evidence_buffer_size ||
+        (!endorsements_buffer && endorsements_buffer_size) ||
+        (endorsements_buffer && !endorsements_buffer_size))
         OE_RAISE(OE_INVALID_PARAMETER);
 
-    plugin_node = oe_attest_find_plugin(verifiers, &evidence->format_id, NULL);
+    if (!format_id)
+    {
+        oe_attestation_header_t* evidence =
+            (oe_attestation_header_t*)evidence_buffer;
+
+        if (evidence_buffer_size < sizeof(oe_attestation_header_t) ||
+            evidence->version != OE_ATTESTATION_HEADER_VERSION)
+            OE_RAISE(OE_INVALID_PARAMETER);
+
+        if (endorsements_buffer)
+        {
+            oe_attestation_header_t* endorsements =
```

Comment:
It is unnecessary for the endorsements be wrapped around oe_attestation_header_t.

Reply:
This is an existing behavior, and there is existing ctest code to specifically verify that the headers in the evidence and endorsements have matching format IDs.

The current proposal does not try to change this existing behavior.

Reply:
Opened issue [3241](https://github.com/openenclave/openenclave/issues/3241) for further investigation.